pkg/prom/diagnostics.go

```diff
@@ -107,6 +119,24 @@ var diagnosticDefinitions map[string]*diagnosticDefinition = map[string]*diagnos
 		Description: "If the 'kubecost_container_cpu_usage_irate' recording rule is not set up, Allocation pipeline build may put pressure on your Prometheus due to the use of a subquery.",
 		DocLink:     "https://docs.kubecost.com/install-and-configure/install/custom-prom",
 	},
+	CAdvisorWorkingSetBytesMetricID: {
+		ID:          CAdvisorWorkingSetBytesMetricID,
+		QueryFmt:    `absent_over_time(container_memory_working_set_bytes{container="cost-model", container!="POD", instance!=""}[5m] %s)`,
+		Label:       "cAdvsior working set bytes metrics available",
+		Description: "Determine if cAdvisor working set bytes metrics are available during last 5 minutes.",
+	},
+	KSMCPUCapacityMetricID: {
+		ID:          KSMCPUCapacityMetricID,
+		QueryFmt:    `avg_over_time(kube_node_status_capacity_cpu_cores[5m] %s) == 0`,
```

Comment:
@nik-kc a lot of ideas being thrown around on this parent issue! I am wondering whether you want to check absent_over_time here rather than an average of 0? My interpretation is that if KSM is not emitting the metric, something is broken. 

In the same vein, couldn't we have a fully allocated node where everything is functioning? That would have a 0 allocatable core result, but would not necessarily be an error. 

Would be interested to hear your thoughts in favor of checking for a 0 value though!

Reply:
Good catch! My thoughts were, I'm still new to Prometheus and its functions and their interactions with scalar vs. vector values, so this is what I got working at the time. But your ideas make much more sense! Keeps them consistent with the other diagnostic queries.

Feedback like this is invaluable, so thank you for taking a look!